### DISCUSSION TITLE
Compile plugin loadPlugins cannot get options

### DISCUSSION THREAD

**Discussion by jxnu-liguobin** [scala/scala3#17238] [Compiler internals Q&A] [answered]
https://github.com/lampepfl/dotty/blob/e25362d9ce1ff5ba6bde92803fbfb231fae0de7e/compiler/src/dotty/tools/dotc/plugins/Plugins.scala#L94 

error

```
scalac: bad option: -P:opt2
```
My config: 
RollsCompilerPlugin is a StandardPlugin
```
    scalacOptions ++= Seq(
      s"-P:RollsCompilerPlugin:opt1,opt2"
    )
```

**Comment by jxnu-liguobin** [comment]
work
```
-P:RollsCompilerPlugin:opt1,
-P:RollsCompilerPlugin:opt2
```

but compiler comment is 
``` scala
  def addPluginPhases(plan: List[List[Phase]])(using Context): List[List[Phase]] = {
    // plugin-specific options.
    // The user writes `-P:plugname:opt1,opt2`, but the plugin sees `List(opt1, opt2)`.
    def options(plugin: Plugin): List[String] = {
      def namec = plugin.name + ":"
      ctx.settings.pluginOptions.value filter (_ startsWith namec) map (_ stripPrefix namec)
    }
``` 

😅

**Reply by ckipp01** [reply, marked as answer]
Hey @jxnu-liguobin, thanks for reporting this and sorry this has sat here for such a long time. This does indeed seem to be a mismatch  between what the comment says and what we have in the code. I've created https://github.com/lampepfl/dotty/pull/17448 to either address this or get some clarity.
